src/Microsoft.DotNet.Wpf/src/PresentationFramework/System/Windows/Documents/WinRTSpellerInteropExtensions.cs

```diff
@@ -0,0 +1,191 @@
+        /// the tokens to account for any missed text "in-between" those tokens. 
+        /// </summary>
+        /// <param name="segmenter">Word-breaker instance</param>
+        /// <param name="text">The text being tokenized</param>
+        /// <param name="spellChecker">The spell-checker instance used to augment the tokenizing process</param>
+        /// <param name="owner">Calling <see cref="WinRTSpellerInterop"/> instance</param>
+        /// <returns></returns>
+        /// <remarks>
+        /// Windows.Data.Text.WordsSegmenter tends to drop punctuation characters like period ('.') 
+        /// when tokenizing text. Though this behavior is compatible with a vast majority of text-processing
+        /// scenarios (like word-counting), it is not ideal for spell-checking. 
+        /// 
+        /// In this method, the following <paramref name="spellChecker"/> augmented heuristic is applied to update the token-list generated by 
+        /// <paramref name="segmenter"/>. 
+        /// 
+        ///  - Identify if any text 'missingFragment' has been dropped by the <paramref name="segmenter"/>
+        ///  - If the token immediately preceding 'missingFragment', previousToken, has a spelling error, then attempt to 
+        ///     create new candiate tokens in the following order:
+        ///     
+        ///             previousToken + missingFragment[0..0]
+        ///             previousToken + missingFragment[0..1]
+        ///             previousToken + missingFragment[0..2]
+        ///             ...
+        ///             ...
+        ///             previousToken + missingFragment[0..LEN-1], where LEN = LEN(missingFragment)
+        ///             
+        ///  - Select the first candidate token that is free of spelling errors, and replace 'previousToken' with it. 
+        ///  - For performance reasons, we choose a constant MAXLEN = 4 such that when LEN > MAXLEN, only MAXLEN
+        ///     tokens are considered. 
+        ///     - MAXLEN = 4 is a somewhat arbitrary choice, though it seems more than sufficient to address common 
+        ///       problems this heuristic is intended to help with. 
+        ///       
+        ///     - Typical word-breaking problems that have been observed empirically involve only one missed character,
+        ///       for which MAXLEN=1 would be sufficient. MAXLEN=4 is chosen as a sufficiently-large tradeoff between
+        ///       correctness and performance. 
+        ///       
+        ///     - Also see https://github.com/dotnet/wpf/pull/2753#issuecomment-602120768 for a discussion related to this. 
+        /// </remarks>
+        public static IReadOnlyList<SpellerSegment> ComprehensiveGetTokens(
+            this WordsSegmenter segmenter,
+            string text,
+            SpellChecker spellChecker,
+            WinRTSpellerInterop owner)
+        {
+            IReadOnlyList<WordSegment> tokens = segmenter?.GetTokens(text) ?? Array.Empty<WordSegment>();
```

Comment:
Fx uses new List<T>().AsReadOnly, where here you use Array.Empty<T>.  What's the difference?  Both are available in both platforms, so any reason to prefer one over the other should apply in both platforms.

Reply:
Array.Empty<T> has been available since .NET 4.6. Our internal NetFx builds seem to build against even older System.Runtime ref-assemblies, and refuse to compile uses of Array.Empty<T>. 

Array.Empty<T> is a little bit better, since it it doesn't have to instantiate a ReadOnlyCollection instance. When its known that a 0-sized IReadOnlyList is needed, it's a bit more efficient this way.

Reply:
Also see community feedback on https://github.com/dotnet/wpf/pull/2871